### PR TITLE
Bump Dekorate to 3.4.0

### PR DESCRIPTION
**Pull request by Sgitario**
Changes: https://github.com/dekorateio/dekorate/releases/tag/3.4.0

This version includes a detecion cycle logic to spot issues when applying decorators. 

Actually, we have detected a couple of issues with the decorators:
- RemoveDeploymentTriggerDecorator:

```
[error]: Build step io.quarkus.kubernetes.deployment.KubernetesProcessor#build threw an exception: java.lang.RuntimeException: Cycle detected when ordering decorators: 
- io.quarkus.kubernetes.deployment.ChangeDeploymentTriggerDecorator
- io.quarkus.kubernetes.deployment.RemoveDeploymentTriggerDecorator
```

- RemoveOptionalFromConfigMapEnvSource, ...

```
[error]: Build step io.quarkus.kubernetes.deployment.KubernetesProcessor#build threw an exception: java.lang.RuntimeException: Cycle detected when ordering decorators: 
- io.quarkus.kubernetes.deployment.RemoveOptionalFromSecretEnvSourceDecorator
- io.quarkus.kubernetes.deployment.RemoveOptionalFromSecretKeySelectorDecorator
- io.quarkus.kubernetes.deployment.RemoveOptionalFromConfigMapEnvSourceDecorator
- io.quarkus.kubernetes.deployment.RemoveOptionalFromConfigMapKeySelectorDecorator
```

### CODE REVIEW

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RemoveOptionalFromConfigMapEnvSourceDecorator.java [hunk 1]
```diff
@@ -14,6 +18,8 @@ public void andThenVisit(ConfigMapEnvSourceFluent ref) {
 
     @Override
     public Class<? extends Decorator>[] after() {
-        return new Class[] { AddEnvVarDecorator.class, ApplicationContainerDecorator.class };
+        return new Class[] { AddEnvVarDecorator.class,
+                AddSecretVolumeDecorator.class, AddSecretVolumeToRevisionDecorator.class,
+                AddConfigMapVolumeToRevisionDecorator.class, AddConfigMapVolumeDecorator.class };
```
Comment (Sgitario) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RemoveOptionalFromConfigMapEnvSourceDecorator.java:23]: To fix the cycle issue, I realized that there were no ApplicationContainerDecorator types that adds this optional flag. However, I added the decorators that do set this value instead.

**Review by geoand — APPROVED** [review]

**Review by iocanel — APPROVED** [review]
lgtm

**Comment by Sgitario** [comment]
I've updated the pull request to include the revert of https://github.com/quarkusio/quarkus/pull/30849. FYI: @gsmet

**Comment by gsmet** [comment]
@Sgitario should we backport this to 2.16 too?

**Comment by Sgitario** [comment]
> @Sgitario should we backport this to 2.16 too?

No, this is an important change that might have breaking changes

**Pull request merged**